src/v/cluster/topic_table_probe.cc

```diff
@@ -9,14 +9,72 @@
 
 #include "cluster/topic_table_probe.h"
 
+#include "cluster/logger.h"
 #include "cluster/topic_table.h"
 #include "config/configuration.h"
+#include "config/node_config.h"
 #include "prometheus/prometheus_sanitize.h"
 
 namespace cluster {
 
 topic_table_probe::topic_table_probe(const topic_table& topic_table)
-  : _topic_table(topic_table) {}
+  : _topic_table(topic_table) {
+    setup_metrics();
+}
+
+void topic_table_probe::setup_metrics() {
+    if (
+      config::shard_local_cfg().disable_public_metrics()
+      || ss::this_shard_id() != 0) {
+        return;
+    }
+    namespace sm = ss::metrics;
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("node:movements"),
+      {
+        sm::make_gauge(
+          "moving_to_partitions",
+          [this] { return moving_to_partitions; },
+          sm::description("Amount of partitions that is moving to node")),
+        sm::make_gauge(
+          "moving_from_partitions",
+          [this] { return moving_from_partitions; },
+          sm::description("Amount of partitions that is moving from node")),
+      });
+}
+
+void topic_table_probe::handle_update_addition(const model::ntp& ntp) {
+    if (_topic_table.moving_from_node(
+          config::node().node_id(),
+          ntp,
+          _topic_table._updates_in_progress.at(ntp))) {
```

Comment:
nit: can be just a single lookup (in the other function as well)

Reply:
fixed